x-pack/plugins/dashboard_mode/server/dashboard_mode_request_interceptor.js

```diff
@@ -25,9 +27,27 @@ export function createDashboardModeRequestInterceptor(dashboardViewerApp) {
     type: 'onPostAuth',
     async method(request, h) {
       const { auth, url } = request;
+      const user = auth.credentials;
```

Comment:
As you can tell from the commit history in this PR, the ordering of operations in this function is rather fragile. 

Previously, this logic was part of `dashboard_mode_auth_scope`, which ran after a successful authentication. Now, this runs as part of "onPostAuth", which runs after auth, but also runs when security is disabled completely.

The existing (and current) logic uses the UI Settings service to retrieve the list of dashboard-only-mode roles. The act of retrieving this setting ends up creating the `config` document if it doesn't already exist. Various functional tests have come to indirectly rely on this behavior, so I've opted to maintain that in this PR. I had originally hoped to optimize the existing implementation by not querying the UI Settings Servive unless strictly necessary, but I wasn't able to do so without also revisiting all of the existing functional tests. I vote to work in that when we remove dashboard-only-mode altogether.

Reply:
> How many (approximately) of such weird tests we have? I'm fine with leaving it as is, just curious.

It's hard to quantify this since the FTR doesn't run test suites to completion once it finds a test failure. I know we had failures in infra and canvas, but I'm sure there are a number of other impacted suites.

Reply:
Okay, thanks, that's what I wanted to know. I was just curious whether tests were failing within one "area" or not.